internal/versions/api.go

```diff
@@ -107,55 +229,52 @@ func (h *apiHandler) V2ListSupportedOpenshiftVersions(ctx context.Context, param
 			}
 		}
 
-		for _, arch := range supportedArchs {
-			key := *releaseImage.OpenshiftVersion
+		supportLevel, err := getReleaseImageSupportLevel(releaseImage)
+		if err != nil {
+			h.log.Debug("error occurred while trying to get the support level of release image version '%s'", *releaseImage.Version)
+			continue
+		}
+
+		for _, arch := range releaseImage.CPUArchitectures {
+			displayName := *releaseImage.Version
 			if arch == "" {
 				// Empty implies default architecture
 				arch = common.DefaultCPUArchitecture
 			}
 
 			// In order to mark a specific version and architecture as supported we do not
 			// only need to have an available release image, but we need RHCOS image as well.
-			if _, err := h.osImages.GetOsImage(key, arch); err != nil {
-				h.log.Debugf("Marking architecture %s for version %s as not available because no matching OS image found", arch, key)
+			if _, err := h.osImages.GetOsImage(displayName, arch); err != nil {
+				h.log.Debugf("Marking architecture %s for version %s as not available because no matching OS image found", arch, displayName)
 				continue
 			}
 
 			// In order to handle multi-arch release images correctly in the UI, we need to tune
 			// their DisplayName to contain an appropriate suffix. If the suffix comes from the
 			// JSON definition we do nothing, but in case it's missing there (because CVO is not
 			// reporting it), we add it ourselves.
-			displayName := *releaseImage.Version
-			if len(supportedArchs) > 1 && !strings.HasSuffix(displayName, "-multi") {
+			if len(releaseImage.CPUArchitectures) > 1 && !strings.HasSuffix(displayName, "-multi") {
 				displayName = displayName + "-multi"
 			}
 
-			openshiftVersion, exists := openshiftVersions[key]
+			openshiftVersion, exists := openshiftVersions[displayName]
```

Comment:
This changes the key from `releaseImage.OpenshiftVersion` to `releaseImage.Version`.
Was that intentional?

Reply:
Yes, because `openshiftVersion` may be x.y.z or x.y while `Version` is always x.y.z, there are now two different versions with the same `openshiftVersion`